PolicyDecisionPoint/src/main/java/it/cnr/iit/ucs/pdp/PolicyDecisionPoint.java

```diff
@@ -268,31 +238,21 @@ private AbstractResult evaluateContext( EvaluationCtx context, PolicyFinder poli
         PolicyFinderResult finderResult = policyFinder.findPolicy( context );
 
         if( finderResult.notApplicable() ) {
-            return ResultFactory.getFactory()
-                .getResult( AbstractResult.DECISION_NOT_APPLICABLE, context );
-        }
-        if( finderResult.indeterminate() ) {
-            return ResultFactory.getFactory().getResult(
-                AbstractResult.DECISION_INDETERMINATE, finderResult.getStatus(),
-                context );
-        }
-        // we found a valid policy, list all found policies if XACML 3.0
-        if( context instanceof XACML3EvaluationCtx
-                && ( (RequestCtx) context.getRequestCtx() ).isReturnPolicyIdList() ) {
+            return ResultFactory.getFactory().getResult( AbstractResult.DECISION_NOT_APPLICABLE, context );
+        } else if( finderResult.indeterminate() ) {
+            return ResultFactory.getFactory().getResult( AbstractResult.DECISION_INDETERMINATE, finderResult.getStatus(), context );
+        } else if( context instanceof XACML3EvaluationCtx && ( (RequestCtx) context.getRequestCtx() ).isReturnPolicyIdList() ) {
```

Comment:
same here. I think we do not need the two "else"

Reply:
ok!